### PR TITLE
Improved coverage

### PR DESCRIPTION
**Pull request by ErikBjare**
Merge after https://github.com/ActivityWatch/aw-core/pull/57

### CODE REVIEW

File: aw_datastore/storages/abstract.py [hunk 1]
```diff
@@ -35,11 +35,11 @@ def get_metadata(self, bucket_id: str) -> dict:
 
     @abstractmethod
     def get_events(self, bucket_id: str, limit: int,
-                   starttime: Optional[datetime]=None, endtime: Optional[datetime]=None) -> List[Event]:
+                   starttime: datetime=None, endtime: datetime=None) -> List[Event]:
         raise NotImplementedError
 
     def get_eventcount(self, bucket_id: str,
-                   starttime: Optional[datetime]=None, endtime: Optional[datetime]=None) -> int:
+                   starttime: datetime=None, endtime: datetime=None) -> int:
```
Comment (johan-bjareholt) [inline aw_datastore/storages/abstract.py]: Why remove Optional?

File: aw_datastore/storages/memory.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ class MemoryStorage(AbstractStorage):
     """For storage of data in-memory, useful primarily in testing"""
     sid = "memory"
 
-    def __init__(self, testing):
+    def __init__(self, testing: bool) -> None:
```
Comment (johan-bjareholt) [inline aw_datastore/storages/memory.py]: Isn't that implicit? init can never return something I believe

File: aw_datastore/storages/abstract.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ def create_bucket(self, bucket_id: str, type_id: str, client: str,
         raise NotImplementedError
 
     @abstractmethod
-    def delete_bucket(self, bucket_id: str) -> None:
+    def delete_bucket(self, bucket_id: str) -> bool:
```
Comment (johan-bjareholt) [inline aw_datastore/storages/abstract.py]: If delete_bucket fails it should not return false, it should raise an exception.
If you get a failure from delete_bucket you also want to know the cause.

File: aw_datastore/storages/abstract.py (hunk 1 above)
Comment (ErikBjare) [reply aw_datastore/storages/abstract.py]: It's implicit when the default argument is None.

File: aw_datastore/storages/memory.py (hunk 2 above)
Comment (ErikBjare) [reply aw_datastore/storages/memory.py]: mypy won't typecheck if the function doesn't have a signature, and it requires __init__ to return None.

File: aw_datastore/storages/abstract.py (hunk 1 above)
Comment (johan-bjareholt) [reply aw_datastore/storages/abstract.py]: Maybe for type hinting, but for actual type checking for mypy that doesn't make sense

File: aw_datastore/storages/memory.py (hunk 2 above)
Comment (johan-bjareholt) [reply aw_datastore/storages/memory.py]: Alright

File: aw_datastore/storages/abstract.py (hunk 1 above)
Comment (ErikBjare) [reply aw_datastore/storages/abstract.py]: It seems that implicit `Optional` is frowned upon (but allowed, for now): https://github.com/python/typing/issues/275

File: aw_datastore/storages/abstract.py (hunk 3 above)
Comment (ErikBjare) [reply aw_datastore/storages/abstract.py]: Fixed

**Comment by johan-bjareholt** [comment]
For a peewee->sqlite database migration to start these are the criterias:
- Start a Datastore object with sqlite backend
- There does not exist a previous sqlite database
- There does exist a previous peewee.v2.db database

So the solution to avoid a migration during testing would be to run the datastore tests with sqlite before running them with peewee.

**Pull request merged**